executor/builder.go

```diff
@@ -2261,16 +2265,54 @@ func (builder *dataReaderBuilder) buildUnionScanForIndexJoin(ctx context.Context
 	return us, err
 }
 
-func (builder *dataReaderBuilder) buildTableReaderForIndexJoin(ctx context.Context, v *plannercore.PhysicalTableReader, lookUpContents []*indexJoinLookUpContent) (Executor, error) {
+func (builder *dataReaderBuilder) buildTableReaderForIndexJoin(ctx context.Context, v *plannercore.PhysicalTableReader,
+	lookUpContents []*indexJoinLookUpContent, innerJoinKeys []expression.Expression) (Executor, error) {
 	e, err := buildNoRangeTableReader(builder.executorBuilder, v)
 	if err != nil {
 		return nil, err
 	}
-	handles := make([]int64, 0, len(lookUpContents))
+	handles := make([]types.Datum, 0, len(lookUpContents))
 	for _, content := range lookUpContents {
-		handles = append(handles, content.keys[0].GetInt64())
+		handles = append(handles, content.keys[0])
 	}
-	return builder.buildTableReaderFromHandles(ctx, e, handles)
```

Comment:
Why we don't use this method?

Reply:
Because we need recalculate the range of handles by `expr.ReverseEval`. All we need is
```
[ReverseEval(handles[i], Floor), ReverseEval(handles[i], Ceiling)]
```
instead of 
```
[handles[i], handles[i]]
```
So we don't need this method.